src/components/tl/mlx5/tl_mlx5_team.c

```diff
@@ -92,7 +95,9 @@ UCC_CLASS_CLEANUP_FUNC(ucc_tl_mlx5_team_t)
     tl_debug(self->super.super.context->lib, "finalizing tl team: %p", self);
 
     ucc_tl_mlx5_dm_cleanup(self);
-    ucc_tl_mlx5_alltoall_cleanup(self);
+    if (self->a2a_state != TL_MLX5_TEAM_STATE_ALLTOALL_NOT_AVAILABLE) {
```

Comment:
just to make sure: was this necessary? I thought I made sure this was ok. Have you check there is no case of memory leak here if we just skip the cleanup?

Reply:
No, I have not checked that. However, during team create, if the state becomes not-available, we already have called the cleanup function over there (given the fact that local a2a is READY but for some reasons, some of the process are not READY). So that is why we are avoiding the cleanup again. But if you think it is OK to do cleanup again, I can remove this check here.